### PR TITLE
Add upload job to analysis in trailblazer

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR ensures the rsync jobs are associated with their corresponding analysis in trailblazer.
Part of the [implementation plan](https://github.com/Clinical-Genomics/cigrid-ui/issues/492#issuecomment-1905771598) for https://github.com/Clinical-Genomics/cigrid-ui/issues/492.

### Added
- Associate slurm rsync job with analysis in trailblazer

### This [version](https://semver.org/) is a
- [ ] **MINOR**

### CODE REVIEW

**Review by diitaz93 — APPROVED** [review]
👍

File: cg/apps/tb/api.py [hunk 1]
```diff
@@ -148,3 +149,12 @@ def add_comment(self, case_id: str, comment: str):
         self.query_trailblazer(
             command="add-comment", request_body=request_body, method=APIMethods.PUT
         )
+
+    def add_upload_job_to_analysis(self, analyis_id: str, slurm_job_id: str) -> None:
+        create_request = CreateJobRequest(slurm_id=slurm_job_id, job_type="upload")
```
Comment (henrikstranneheim) [inline cg/apps/tb/api.py:154]: Would be create with a enum class for job type.

File: cg/apps/tb/dto/create_job_request.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from pydantic import BaseModel
+
+
+class CreateJobRequest(BaseModel):
+    slurm_id: str
+    job_type: str
```
Comment (henrikstranneheim) [inline cg/apps/tb/dto/create_job_request.py:6]: Let's restrict these to the same as defined in TB.

**Review by henrikstranneheim — APPROVED** [review]
👍

File: cg/cli/upload/clinical_delivery.py [hunk 3]
```diff
@@ -73,6 +75,7 @@ def upload_clinical_delivery(context: click.Context, case_id: str, dry_run: bool
             data_analysis=Pipeline.RSYNC,
             ticket=case.latest_ticket,
         )
+        trailblazer_api.add_upload_job_to_analysis(analysis_id=analysis.id, job_id=job_id)
```
Comment (Vince-janv) [inline cg/cli/upload/clinical_delivery.py:78]: Good idea to keep `add_pending_analysis` for now and remove it later 👍

**Review by Vince-janv — APPROVED** [review]
⭐

File: cg/apps/tb/dto/create_job_request.py (hunk 2 above)
Comment (seallard) [reply cg/apps/tb/dto/create_job_request.py:6]: The `slurm_id` exists and `job_type` will exist once I merge a PR 👍

Comment (seallard) [reply cg/apps/tb/dto/create_job_request.py:6]: Added enum 👍

**Pull request merged**